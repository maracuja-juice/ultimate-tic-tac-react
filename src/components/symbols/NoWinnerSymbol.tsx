import * as React from 'react';
import './symbols.css';
import './symbolAnimations.css';
import * as classNames from 'classnames';

interface NoWinnerSymbolProps {
}

interface NoWinnerSymbolState {
}

export class NoWinnerSymbol extends React.Component<NoWinnerSymbolProps, NoWinnerSymbolState> {

    constructor( props: NoWinnerSymbolProps ) {
        super( props );
    }

    render() {
        var iconClass = classNames({
                                       'material-icons': true,
                                       'big-symbol': true,
                                       'no-winner': true,
                                       'animate-no-winner-big-symbol': true
                                   });
        return (
            <i className={iconClass}>remove</i>
        );
    }
}