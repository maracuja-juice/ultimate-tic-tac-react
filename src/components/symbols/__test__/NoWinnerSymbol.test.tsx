import * as React from 'react';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import { NoWinnerSymbol } from '../NoWinnerSymbol';

configure( {adapter: new ReactSixteenAdapter()} );

describe( 'NoWinnerSymbol', () => {
    it( 'should not explode', () => {
        const component = shallow( <NoWinnerSymbol/> );
        expect( component ).not.toBeNull();
    } );

    it( 'should match snapshot', () => {
        const noWinnerSymbol = shallow( <NoWinnerSymbol/> );
        expect( noWinnerSymbol ).toMatchSnapshot();
    } );

    it( 'should have i element with some kind of icon', () => {
        const component = shallow( <NoWinnerSymbol/> );
        expect( component.find( 'i' ) ).toHaveLength( 1 );
        expect( component.find( 'i' ).text() ).not.toBeUndefined();
    } );

    it( 'should have icon-no-winner, no-winner and big-symbol class by default', () => {
        const component = shallow( <NoWinnerSymbol/> );
        expect( component.find( 'i' ).hasClass('icon-no-winner')).toBe(true);
        expect( component.find( 'i' ).hasClass('big-symbol')).toBe(true);
        expect( component.find( 'i' ).hasClass('no-winner') ).toBe(true);
    } );
} );