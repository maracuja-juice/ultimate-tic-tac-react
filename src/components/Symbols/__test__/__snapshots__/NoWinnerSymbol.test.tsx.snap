// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoWinnerSymbol should match snapshot, animate false 1`] = `
<i
  aria-label="No Winner Symbol"
  className="icon-no-winner big-symbol no-winner"
/>
`;

exports[`NoWinnerSymbol should match snapshot, animate true 1`] = `
<i
  aria-label="No Winner Symbol"
  className="icon-no-winner big-symbol no-winner animate-no-winner"
/>
`;
